starter/source/general_controls/damping/dampdtnoda.F

```diff
@@ -0,0 +1,120 @@
+Copyright>        MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+Copyright>        GNU Affero General Public License for more details.
+Copyright>    
+Copyright>        You should have received a copy of the GNU Affero General Public License
+Copyright>        along with this program.  If not, see <https://www.gnu.org/licenses/>.
+Copyright>    
+Copyright>    
+Copyright>        Commercial Alternative: Altair Radioss Software 
+Copyright>    
+Copyright>        As an alternative to this open-source version, Altair also offers Altair Radioss 
+Copyright>        software under a commercial license.  Contact Altair to discuss further if the 
+Copyright>        commercial version may interest you: https://www.altair.com/radioss/.    
+Chd|====================================================================
+Chd|  DAMPDTNODA                  source/general_controls/damping/hm_read_damp.F
+Chd|-- called by -----------
+Chd|        LECTUR                        source/starter/lectur.F       
+Chd|-- calls ---------------
+Chd|====================================================================
+      SUBROUTINE DAMPDTNODA(MS ,IN  ,STIFN ,STIFR ,IGRNOD   ,DAMPR )
+C-----------------------------------------------
+C   M o d u l e s
+C-----------------------------------------------
+      USE GROUPDEF_MOD
+C-----------------------------------------------
+C   I m p l i c i t   T y p e s
+C-----------------------------------------------
+#include      "implicit_f.inc"
+C-----------------------------------------------
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "com01_c.inc"
+#include      "com04_c.inc"
+#include      "units_c.inc"
+#include      "param_c.inc"
+C-----------------------------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+C     REAL
+      my_real , DIMENSION(NUMNOD), INTENT(IN) ::MS 
+      my_real , DIMENSION(NUMNOD*IRODDL), INTENT(IN) ::IN 
+      my_real , DIMENSION(NUMNOD), INTENT(INOUT) ::STIFN 
+      my_real , DIMENSION(NUMNOD), INTENT(INOUT) ::STIFR 
+      my_real , DIMENSION(NRDAMP,NDAMP), INTENT(IN) ::DAMPR 
+C-----------------------------------------------
+      TYPE (GROUP_)  , DIMENSION(NGRNOD) :: IGRNOD
+C-----------------------------------------------
+C   L o c a l   V a r i a b l e s
+C-----------------------------------------------
+      INTEGER N,I,J,ISK,IGR,ND,K,L,LL
+      my_real FACTB,DAMPBR,DAMPAR,D_TSTART,D_TSTOP,DAMPT,
+     .   DT2N(NUMNOD),BB,COEFF,DTN,DTN2,DAMPAI,DAMPBI,DT0,DAMPA3,
+     .   ALPHA05,BETA1,DTMAX,FAC,DTMI1,FAC0    
+C----------------------------------------------------------
+        CALL READ_ENGINE_DTMIN(DTMI1)
+        DTMAX = EP04
+        DT0 = DTMAX
+        DO I=1,NDAMP
+C-----traited before        
+          IF (DAMPR(19,I)/=0) CYCLE	   	        	      	      	              
```

Comment:
is it possible to avoid using CYCLE here ( cf prog rules)

Reply:
for me, it's not